### ISSUE TITLE
Hub does not currently emit an event when a refresh token expires or access token expires

### ISSUE THREAD

**Issue by mrSingh007** [aws-amplify/amplify-js#12812]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

Angular

### Amplify APIs

Authentication

### Amplify Version

v6

### Amplify Categories

auth

### Backend

None

### Environment information

<details>

```
# Put output below this line
OS: MacOS 13.5.2
CPU: (8) arm64 Apple M1 Pro
Memory: 51.52 MB / 16.00 GB

Node: 16.20.0
npm: 8.19.4

@aws-amplify/auth: ^6.0.10
@aws-amplify/auth: ^6.0.10
aws-amplify: ^6.0.10
```

</details>

### Describe the bug

When access or refresh token is expired, no any event is sent to channel 'auth'.  Page need to refresh manually to get new access token.

### Expected behavior

When ever refresh token or access token expired, Hub should receive an event for 'auth' channel. 

### Reproduction steps

install the latest versions as mentioned above.
Listen to events as following: 
```
Hub.listen('auth', (msg:any) => { console.log('Event : ', msg) });
```

Used config:
```
Amplify.configure({
  Auth:{
    Cognito:{
       userPoolId:'<id>',
       userPoolClientId: '<id>',
       loginWith: {
           oauth: {
            domain:'<domain>'
            redirectSignIn: ['<url>'],
            redirectSignOut: ['<url>'],
            scopes: ['openid'],
            responseType: 'token'
           }
        }
      } 
   }
})
```

### Code Snippet

```javascript
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line


```

</details>

**Comment by mrSingh007** [comment]
Seems there is no counter or task running that checks the expiry time of token and then dispatch the event.

**Comment by israx** [comment]
hello @mrSingh007 . Thank you for bringing this up. We currently have a fix for this issue which needs to be released.

**Comment by israx** [comment]
Hello @mrSingh007 . The latest [release](https://www.npmjs.com/package/aws-amplify) included the fix, you should be able to get the hub event when the `refresh_token` is expired

**Comment by mrSingh007** [comment]
I have tested now in 5.3.15 and in version 6.0.12 of 'aws-amplify' and the issue is still there. No any event was dispatched, when token is expired.

**Comment by elorzafe** [comment]
@mrSingh007 you want an specific event when access or id token are expired? What is the use case?

Thanks!

**Comment by elorzafe** [comment]
@mrSingh007 you should be able to see `signOut` event when the `refresh_token` is expired. The library emits this event when there is an API call that tries to pull tokens (e.g. `fetchAuthSession` API). The library cannot detect if the `refresh_token` is expired without doing an operation with Cognito service.

**Comment by cwomack** [comment]
@mrSingh007, did you get a chance to see @elorzafe's comment above?  We'd like to know if there's additional context/use cases around this feature that we could work on.  Would having just the expiration of the access token emit an event be enough here?

**Comment by cwomack** [comment]
Hey, @mrSingh007 👋.  It looks like you may have had a similar question/issue resolved in #12832.  Since we haven't heard back from you on this issue, we'll close this out for now.  Feel free to reply back if there's still a question or blocker here, thanks!
